### PR TITLE
Use auth to decrypt signature and use container name for audience

### PR DESCRIPTION
**Pull request by Tonewall**
Backporting parts of #9325 and #9348 

### Issue describing the changes in this PR
Uses Container name as the valid audience for placeholders in Linux Consumption because sitename are not present when in placeholder mode. This PR also includes changes to accept signature as part of the token for container assignment.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs [hunk 1]
```diff
@@ -30,57 +30,68 @@ public static class ScriptJwtBearerExtensions
...
-                                OnMessageReceived = c =>
-                                {
-                                    // By default, tokens are passed via the standard Authorization Bearer header. However we also support
-                                    // passing tokens via the x-ms-site-token header.
-                                    if (c.Request.Headers.TryGetValue(ScriptConstants.SiteTokenHeaderName, out StringValues values))
-                                    {
-                                        // the token we set here will be the one used - Authorization header won't be checked.
-                                        c.Token = values.FirstOrDefault();
-                                    }
-
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs]: Looks like you're removing code formatting here and below. Why?

Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs]: Oh, I was just cherry-picking from the hotfix PR, but I will correct it

Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs]: Fixed!

**Comment by mathewc** [comment]
The description of this PR should be updated so it describes the actual change being made. Also, it doesn't seem you're linking to the actual changes you're backporting from. I think you're trying to backport a portion of https://github.com/Azure/azure-functions-host/pull/9348?

**Comment by Tonewall** [comment]
> The description of this PR should be updated so it describes the actual change being made. Also, it doesn't seem you're linking to the actual changes you're backporting from. I think you're trying to backport a portion of #9348?

Updated the PR name and linked the PRs backporting from

File: src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs [hunk 2]
```diff
@@ -80,7 +80,25 @@ public static AuthenticationBuilder AddScriptJwtBearer(this AuthenticationBuilde
                             }
                         });
 
-        private static TokenValidationParameters CreateTokenValidationParameters()
+        private static string[] GetValidAudiences()
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs:83]: We should add test coverage for this. The source PR for v4 added test coverage for this. Can we do the same?

Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs:83]: On it

Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs:83]: Added the unit test 👍

**Review by fabiocav — CHANGES_REQUESTED** [review]
Can you please make sure all PR details and information about the change are present?

**Review by mathewc — DISMISSED** [review]

**Comment by Tonewall** [comment]
> Can you please make sure all PR details and information about the change are present?

Filled out the PR description with the changes in the PR

**Comment by Tonewall** [comment]
@fabiocav could you have another look?

**Review by fabiocav — COMMENTED** [review]
Removing change request, but would like to make sure @mathewc approves as well

**Review by fabiocav — APPROVED** [review]

**Review by mathewc — APPROVED** [review]

**Pull request merged**